### ISSUE TITLE
Unresolved symbol with :refer :all

### ISSUE THREAD

**Issue by l3nz** [borkdude/clj-kondo#342]
**version**

[clj-kondo "2019.07.05-alpha"]

**platform**

JVM library, [org.clojure/clojure "1.9.0"]

**problem**

I have a require with `[compojure.core :refer :all]`(I know it's not elegant, but it's allowed). When I call any method, I get it as:

```
src/kpl/http/core.clj:321:4: info: unresolved symbol defroutes
src/kpl/http/core.clj:323:6: info: unresolved symbol GET
src/kpl/http/core.clj:326:6: info: unresolved symbol POST 
```

**expected behavior**

Either it should raise a linting error because I use a `:refer :all` as bad practice, or it should avoid complaining of undefined symbols.

**Comment by borkdude** [comment]
@l3nz clj-kondo uses the `ns` form to resolve symbols, so when you use `:refer :all` it doesn't know these symbols. Given that ClojureScript doesn't support  `:refer :all` and the Clojure style guide recommends `:as` or `:refer [...]` over `:refer :all`, I recommend using those forms. Clj-kondo won't support resolving symbols using `:use` or `:refer :all`.

**Comment by l3nz** [comment]
Makes sense, but in this case it should raise it as an issue (`not parsing because you are a dunce who uses :refer :all`) not fail on non-imported symbols 😄

**Comment by borkdude** [comment]
@l3nz Good idea. I was going to fix this under the umbrella of "how to ns": https://github.com/borkdude/clj-kondo/issues/229.

**Comment by borkdude** [comment]
@l3nz I've decided to give it a try anyway. Now it looks like this:

![Screenshot 2019-07-21 21 31 06](https://user-images.githubusercontent.com/284934/61596072-1511a980-abff-11e9-9326-87b9fb211d25.png)

You can try this with commit https://github.com/borkdude/clj-kondo/commit/92e4247eed9585969d4da348c0e90625b2ceae98.

**Comment by borkdude** [comment]
TODO:
- [ ] fix order (using foo before definition should be unresolved)
- [ ] write test for cljc (using slurp in cljc should be unresolved because of cljs)

**Comment by borkdude** [comment]
@l3nz Final result:

<img width="1122" alt="compojure-config" src="https://user-images.githubusercontent.com/284934/61795274-4d91cd00-ae23-11e9-982a-12869712f16c.png">

**Comment by l3nz** [comment]
Looks wonderful, but as an *opinionated* linter, I would in any case earmark the `:refer :all` with a warning. 🥇

**Comment by borkdude** [comment]
@l3nz: final final result ;)

<img width="1124" alt="compojure-config" src="https://user-images.githubusercontent.com/284934/61805136-863aa200-ae35-11e9-8c61-2c0585944f0c.png">

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/7ee6199b93528701b2bfe5b7a8ddcbff96a0b956
